crates/core/src/config.rs

```diff
@@ -1,32 +1,54 @@
+use alloy::primitives::{Bytes, ChainId};
+use starknet::providers::Url;
 use std::env;
 use tokio::sync::OnceCell;
 
 pub static CONFIG: OnceCell<Config> = OnceCell::const_new();
 
+#[derive(Debug)]
 pub struct Config {
-    pub chain_id: u64,
-    pub rpc_url: String,
+    pub chain_id: ChainId,
+    pub rpc_url: Url,
     pub rpc_chunk_size: u64,
-    pub datalakes: String,
-    pub tasks: String,
+    pub datalakes: Bytes,
+    pub tasks: Bytes,
 }
 
 impl Config {
     pub async fn init(
-        cli_rpc_url: Option<String>,
-        cli_datalakes: Option<String>,
-        cli_tasks: Option<String>,
-        cli_chain_id: Option<u64>,
+        cli_rpc_url: Option<Url>,
+        cli_datalakes: Option<Bytes>,
+        cli_tasks: Option<Bytes>,
+        cli_chain_id: Option<ChainId>,
     ) -> &'static Self {
-        let chain_id =
-            cli_chain_id.unwrap_or_else(|| env::var("CHAIN_ID").unwrap().parse().unwrap());
-        let rpc_url = cli_rpc_url.unwrap_or_else(|| env::var("RPC_URL").unwrap());
+        let chain_id = cli_chain_id.unwrap_or_else(|| {
```

Comment:
wdyt about passing all of those as cli args
env is smth hidden and can be forgotten easily

Reply:
isn't sufficient to have both option to be support? like prio 1 is argument passing, prio 2 is env
using cli arg was bit inconvientent with repetive jobs

Reply:
yes envs + cli support is ideal sol!